src/ripple/app/misc/impl/ValidatorList.cpp

```diff
@@ -1565,7 +1584,10 @@ ValidatorList::getJson() const
 
         auto appendList = [](PublisherList const& publisherList,
                              Json::Value& target) {
-            target[jss::uri] = publisherList.siteUri;
+            if (!publisherList.sourceUri.empty())
+                target[jss::source_uri] = publisherList.sourceUri;
+            if (!publisherList.sourcePeer.empty())
+                target[jss::source_peer] = publisherList.sourcePeer;
```

Comment:
I had same concern about the API breaking change. Here is an example we can probably follow, https://github.com/ripple/rippled/pull/3770. Note how the config is enabled in unit tests. 
(@undertome I forgot it when we chatted.)

Reply:
What about when this method gets called via `OverlayImpl::processCrawl` -> `OverlayImpl::getUnlInfo`. In the former function the `version` for the response message (`response<json_body>`) is being set to "2". In this case should I use 2 as the API version?

Reply:
Let's say the existing code returns API version 1 response and the new code returns API version 2 response. I think we need to keep both code paths, so that the correct response can be returned. 

If a RPC request has `"api_version": 2`, we should return API version 2 response. 
If a RPC request has `"api_version": 1` or if this field is missing, API version 1 response should be returned. `OverlayImpl::processCrawl` should also return API version 1 response.

I think https://github.com/ripple/rippled/pull/3770 is a good and simple example. Please take a look.

Reply:
That doesn't really answer my question. I'm asking whether the version in the response message sent from the `OverlayImpl` should be treated the same as the RPC `api_version` number. As for as I can tell #3770 doesn't address this specific question. BTW, in `OverlayImpl::processCrawl` the version is being hard coded to `2` regardless of any version info in the request. Is that intended? At @ximinez maybe you can shed some light?

Reply:
"I'm asking whether the version in the response message sent from the OverlayImpl should be treated the same as the RPC api_version number." I think not. Let's see what @ximinez thinks.

Reply:
The UNL version (returned from OverlayImpl) is entirely separate from the API version.

Reply:
@ximinez For lists sent via the `OverlaImpl`, is it ok to just always send the new fields instead of `uri`?

Reply:
I didn't even realize that `crawl` had versioning until now, and the rules don't seem to be documented other than https://xrpl.org/peer-crawler.html#response-format noting the version changed in 1.2.1. Maybe somebody else can weigh in, but my thinking is that if you don't change the version number, just add the new fields without removing any.

Reply:
Ok, done